File: octopoints_flutter/lib/service/TeamService.dart
import 'package:octopoints_flutter/service/model/Team.dart';
import 'package:octopoints_flutter/service/model/User.dart';

abstract class TeamService {
  Future<Team> createTeam(Team team);
  Future<int> updateTeams(List<Team> teams);
  Future<int> deleteTeam(Team team);
  Future<List<Team>> getTeamsByMatchId(int id);
  Future<List<int>> addTeammates(int teamId, List<User> users);
  Future<int> removeTeammates(int teamId, int userId);
  Future<Map<User, bool>> getAvailableTeammates(Team team);
}

File: octopoints-kotlin/app/src/main/java/dev/tpavani/octopoints/service/TeamService.kt
package dev.tpavani.octopoints.service

import dev.tpavani.octopoints.service.model.Team
import dev.tpavani.octopoints.service.model.User

interface TeamService {
    suspend fun createTeam(team: Team): Team
    suspend fun updateTeams(teams: List<Team>)
    suspend fun deleteTeam(team: Team)
    suspend fun getTeamsByMatchId(id: Int): List<Team>
    suspend fun addTeammates(teamId: Int, users: List<User>)
    suspend fun updateTeammates(team: Team)
    suspend fun getAvailableTeammates(team: Team): Map<User, Boolean>

}